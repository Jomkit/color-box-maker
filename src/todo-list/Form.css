.Form-submitBtn {
    background-color: seagreen;
    border-radius: 3px;
    color:antiquewhite;
    margin: 2px;
    padding: 2px 5px;
}

.Form-submitBtn:hover {
    background-color: green;
}