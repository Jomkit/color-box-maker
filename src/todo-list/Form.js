import React, {useState} from 'react'
import "./Form.css"

const Form = ({id, toggleEdit, editTodo}) => {
    const [formData, setFormData] = useState({editTask: ''});
    const handleChange = (e) => {
        const {name, value} = e.target;
        setFormData(data => ({
            ...data, 
            [name]: value
        }));
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        
        editTodo(id, formData.editTask);
        toggleEdit();
        setFormData({editTask: ''})
    }
    
  return (
    <form onSubmit={handleSubmit}>
        <label htmlFor='editTask'>Edit Task: </label>
        <input
            id="editTask"
            type='text'
            name='editTask'
            value={formData.editTask}
            onChange={handleChange}
        />
        <button className='Form-submitBtn'>Submit</button>
    </form>
  )
}

export default Form