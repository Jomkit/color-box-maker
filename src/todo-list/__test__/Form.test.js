import { render } from "@testing-library/react"
import Form from "../Form";

it('should render without crashing', () => { 
    render(<Form />);
})

it('should match snapshot', () => { 
    const { asFragment } = render(<Form />);
    expect(asFragment()).toMatchSnapshot();
})

