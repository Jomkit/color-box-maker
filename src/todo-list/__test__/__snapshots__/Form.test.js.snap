// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="editTask"
    >
      Edit Task: 
    </label>
    <input
      id="editTask"
      name="editTask"
      type="text"
      value=""
    />
    <button
      class="Form-submitBtn"
    >
      Submit
    </button>
  </form>
</DocumentFragment>
`;
